pkg/api/handlers/compat/volumes.go

```diff
@@ -0,0 +1,238 @@
+		utils.Error(w, "Something went wrong.", http.StatusInternalServerError, errors.Wrap(err, "Decode()"))
+		return
+	}
+
+	if len(input.Name) > 0 {
+		volumeOptions = append(volumeOptions, libpod.WithVolumeName(input.Name))
+	}
+	if len(input.Driver) > 0 {
+		volumeOptions = append(volumeOptions, libpod.WithVolumeDriver(input.Driver))
+	}
+	if len(input.Labels) > 0 {
+		volumeOptions = append(volumeOptions, libpod.WithVolumeLabels(input.Labels))
+	}
+	if len(input.DriverOpts) > 0 {
+		parsedOptions, err := parse.VolumeOptions(input.DriverOpts)
+		if err != nil {
+			utils.InternalServerError(w, err)
+			return
+		}
+		volumeOptions = append(volumeOptions, parsedOptions...)
+	}
+	vol, err := runtime.NewVolume(r.Context(), volumeOptions...)
+	if err != nil {
+		utils.InternalServerError(w, err)
+		return
+	}
+	config, err := vol.Config()
+	if err != nil {
+		utils.InternalServerError(w, err)
+		return
+	}
+	volResponse := docker_api_types.Volume{
+		Name:       config.Name,
+		Driver:     config.Driver,
+		Mountpoint: config.MountPoint,
+		CreatedAt:  config.CreatedTime.Format(time.RFC3339),
+		Labels:     config.Labels,
+		Options:    config.Options,
+		Scope:      "local",
+		// ^^ We don't have volume scoping so we'll just claim it's "local"
+		// like we do in the `libpod.Volume.Scope()` method
+		//
+		// TODO: We don't include the volume `Status` or `UsageData`, but both
+		// are nullable in the Docker engine API spec so that's fine for now
+	}
+	utils.WriteResponse(w, http.StatusCreated, volResponse)
+}
+
+func InspectVolume(w http.ResponseWriter, r *http.Request) {
+	var (
+		runtime = r.Context().Value("runtime").(*libpod.Runtime)
+	)
+	name := utils.GetName(r)
+	vol, err := runtime.GetVolume(name)
+	if err != nil {
+		utils.VolumeNotFound(w, name, err)
+		return
+	}
+	volResponse := docker_api_types.Volume{
+		Name:       vol.Name(),
+		Driver:     vol.Driver(),
+		Mountpoint: vol.MountPoint(),
+		CreatedAt:  vol.CreatedTime().Format(time.RFC3339),
+		Labels:     vol.Labels(),
+		Options:    vol.Options(),
+		Scope:      vol.Scope(),
+		// TODO: As above, we don't return `Status` or `UsageData` yet
+	}
+	utils.WriteResponse(w, http.StatusOK, volResponse)
+}
+
+func RemoveVolume(w http.ResponseWriter, r *http.Request) {
+	var (
+		runtime = r.Context().Value("runtime").(*libpod.Runtime)
+		decoder = r.Context().Value("decoder").(*schema.Decoder)
+	)
+	query := struct {
+		Force bool `schema:"force"`
+	}{
+		// override any golang type defaults
+	}
+
+	if err := decoder.Decode(&query, r.URL.Query()); err != nil {
+		utils.Error(w, http.StatusText(http.StatusBadRequest), http.StatusBadRequest,
+			errors.Wrapf(err, "Failed to parse parameters for %s", r.URL.String()))
+		return
+	}
+	name := utils.GetName(r)
+	vol, err := runtime.LookupVolume(name)
+	if err != nil {
+		utils.VolumeNotFound(w, name, err)
+		return
+	}
+	if err := runtime.RemoveVolume(r.Context(), vol, query.Force); err != nil {
```

Comment:
We realized recently that our Force and the Docker engine's Force are actually different things (Docker's behavior seems to be that Force means "do not error on missing volume"; ours means "remove any not-running containers using the volume at the same time")

Reply:
As such, I recommend unconditionally setting Force to false for now.

Reply:
Done, will push an extra commit in a moment. I've also added handling for `force` to alter a 404 to a 204 when the named volume can't be found. That means we don't need to caveat our support or explicitly reject the query parameter when we see it. Take a look at the new diff here where you have a chance, @mheon

Reply:
Well, gocritic hates that code structure. I'll make some changes